### PR TITLE
docs: update flame graph visualization

### PR DESCRIPTION
**Pull request by mdcruz**
**What is this feature?**

Updates to the flame graph docs to include:
- how to configure a flame graph (video TBA)
- when to use a flame graph
- rearranging sections to match other visualization types

**Why do we need this feature?**

To explain to our users how to configure a flame graph easily and when to use it.

**Who is this feature for?**

Grafana OSS and Cloud users

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by imatwawana — COMMENTED** [review]
Thanks for these updates! I've left some suggestions.

File: docs/sources/panels-visualizations/visualizations/flame-graph/index.md [hunk 1]
```diff
@@ -19,61 +19,100 @@ weight: 100
...
+
+## Configure a flame graph visualization
+
+Once you’ve created a [dashboard](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/dashboards/build-dashboards/create-dashboard/), the following video shows you how to configure a flame graph visualization:
+
+<!-- video TBA here -->
+
+## Supported data formats
+
+In order to render the flame graph, you must format the data frame data using a [nested set model](https://en.wikipedia.org/wiki/Nested_set_model).
```
[the remark this answers is not in the document]
Comment (mdcruz) [reply docs/sources/panels-visualizations/visualizations/flame-graph/index.md]: Great, thank you!

File: docs/sources/panels-visualizations/visualizations/flame-graph/index.md [hunk 2]
```diff
@@ -19,61 +19,115 @@ weight: 100
...
+## Configure a flame graph visualization
+
+Once you’ve created a [dashboard](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/dashboards/build-dashboards/create-dashboard/), the following video shows you how to configure a flame graph visualization:
+
+<!-- video TBA here -->
+
+## Supported data formats
 
-![<Flame graph>](/media/docs/grafana/panels-visualizations/flamegraph/screenshot-flamegraph-10.1-flamegraph.png)
+To render a flame graph, you must format the data frame data using a [nested set model](https://en.wikipedia.org/wiki/Nested_set_model).
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/flame-graph/index.md:42]:
```suggestion
To render a flame graph, you must format the data frame data using a _nested set model_.
```

File: docs/sources/panels-visualizations/visualizations/flame-graph/index.md [hunk 3]
```diff
@@ -19,61 +19,115 @@ weight: 100
...
+| Field name | Type   | Description                                                                                                                |
+| ---------- | ------ | -------------------------------------------------------------------------------------------------------------------------- |
+| level      | number | The nesting level of the item. In other words how many items are between this item and the top item of the flame graph.    |
+| value      | number | The absolute or cumulative value of the item. This translates to the width of the item in the graph.                       |
+| label      | string | Label to be shown for the particular item.                                                                                 |
+| self       | number | Self value, which is usually the cumulative value of the item minus the sum of cumulative values of its immediate children. |
+
+### Example
+
+The following table is an example of the type of data you need for a flame graph visualization and how it should be formatted.
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/flame-graph/index.md:57]:
```suggestion
The following table is an example of the type of data you need for a flame graph visualization and how it should be formatted:
```

File: docs/sources/panels-visualizations/visualizations/flame-graph/index.md [hunk 4]
```diff
@@ -19,61 +19,115 @@ weight: 100
...
 
 Grayed-out sections are a set of functions that represent a relatively small value and they are collapsed together into one section for performance reasons.
 
-{{< figure src="/media/docs/grafana/panels-visualizations/flamegraph/screenshot-flamegraph-10.1-tooltip.png" max-width="500px" caption="Hover tooltip" >}}
+{{< figure src="/static/img/docs/flame-graph-panel/flame-graph-mode-dark.png" max-width="650px" alt="A flame graph visualization for a system profile with flame graph mode." >}}
 
 You can hover over a specific function to view a tooltip that shows you additional data about that function, like the function's value, percentage of total value, and the number of samples with that function.
 
-### Drop-down actions
+{{< figure src="/media/docs/grafana/panels-visualizations/flamegraph/screenshot-flamegraph-10.1-tooltip.png" max-width="1025px" alt="A flame graph visualization with a hover tooltip." >}}
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/flame-graph/index.md:82]:
```suggestion
{{< figure src="/media/docs/grafana/panels-visualizations/flamegraph/screenshot-flamegraph-10.1-tooltip.png" max-width="650px" alt="A flame graph visualization with a hover tooltip." >}}
```

**Review by imatwawana — APPROVED** [review]

File: docs/sources/panels-visualizations/visualizations/flame-graph/index.md [hunk 5]
```diff
@@ -19,61 +19,115 @@ weight: 100
...
+#### Time-series table
+
+| level | value    | self   | label                                     |
+| ----- | -------- | ------ | ----------------------------------------- |
+| 0     | 16.5 Bil | 16.5 K | total                                     |
+| 1     | 4.10 Bil | 4.10 k | test/pkg/agent.(\*Target).start.func1     |
+| 2     | 4.10 Bil | 4.10 K | test/pkg/agent.(\*Target).start.func1     |
+| 3     | 3.67 Bil | 3.67 K | test/pkg/distributor.(\*Distributor).Push |
+| 4     | 1.13 Bil | 1.13 K | compress/gzip.(\*Writer).Write            |
+| 5     | 1.06 Bil | 1.06 K | compress/flate.(\*compressor).write       |
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/flame-graph/index.md:68]: Is the word "flate" correct here? It's causing an error so I just want to make sure.
```suggestion
| 5     | 1.06 Bil | 1.06 K | compress/flat.(\*compressor).write       |
```

File: docs/sources/panels-visualizations/visualizations/flame-graph/index.md [hunk 6]
```diff
@@ -19,61 +19,115 @@ weight: 100
 
 # Flame graph
 
-Flame graphs let you visualize profiling data. Using this visualization, the profile can be represented as a flame graph, table, or both
+Flame graphs let you visualize [profiling](https://grafana.com/docs/pyroscope/latest/introduction/profiling/) data. Using this visualization, a profile can be represented as a [flame graph](#flame-graph-mode), [top table](#top-table-mode), or both.
```
Comment (knylander-grafana) [inline docs/sources/panels-visualizations/visualizations/flame-graph/index.md:22]: You might also consider these two additional links: 
* Flame graphs: Visualizing performance data, https://grafana.com/docs/pyroscope/latest/view-and-analyze-profile-data/flamegraphs/
* Understand profile types and their uses in Pyroscope, https://grafana.com/docs/pyroscope/latest/view-and-analyze-profile-data/profiling-types/ (explains the  different profilers)

Comment (mdcruz) [reply docs/sources/panels-visualizations/visualizations/flame-graph/index.md:22]: Thank you, @knylander-grafana. I'll add these links!

**Comment by mdcruz** [comment]
@imatwawana I've added the flame graph video. This PR is ready to be merged when you are ready :)

File: docs/sources/panels-visualizations/visualizations/flame-graph/index.md [hunk 7]
```diff
@@ -19,61 +19,115 @@ weight: 100
...
+{{< figure src="/static/img/docs/flame-graph-panel/flame-graph-dark-mode.png" max-width="1025px" alt="A flame graph visualization for a system profile with both flame graph and top table mode." >}}
 
-A flame graph takes advantage of the hierarchical nature of profiling data. It condenses data into a format that allows you to easily see which code paths are consuming the most system resources, such as CPU time, allocated objects, or space when measuring memory. Each block in the flame graph represents a function call in a stack and its width represents its value.
+You can use a flame graph visualization if you need to:
+
+- Identify any performance hotspots to find where code optimizations may be needed.
+- Diagnose the root cause of any performance degradation.
+- Analyze the behavior of complex systems, including distributed systems or microservices architectures.
+
+To know more about how Grafana Pyroscope visualizes flame graphs, refer to [Flame graphs: Visualizing performance data](https://grafana.com/docs/pyroscope/latest/view-and-analyze-profile-data/flamegraphs/).
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/flame-graph/index.md:34]:
```suggestion
To learn more about how Grafana Pyroscope visualizes flame graphs, refer to [Flame graphs: Visualizing performance data](https://grafana.com/docs/pyroscope/latest/view-and-analyze-profile-data/flamegraphs/).
```

**Pull request merged**